json-schemas/records/records-write.json

```diff
@@ -200,6 +200,36 @@
               ]
             }
           ]
+        },
+        {
+          "$comment": "rule: `recipient` can only exist if `protocol` exists`",
```

Comment:
@thehenrytsai wondering if using `dependencies` might be easier to represent `recipient`'s dependency on `protocol`'s presence? simplified example:

```json
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "$id": "https://identity.foundation/dwn/json-schemas/records-write.json",
  "type": "object",
  "additionalProperties": false,
  "required": [
    "recordId"
  ],
  "properties": {
    "recordId": {
      "type": "string"
    },
    "protocol": {
      "type": "string"
    },
    "recipient": {
      "type": "string"
    }
  },
  "dependencies": {
    "recipient": ["protocol"]
  }
}
```

Reply:
OMG! Thanks @mistermoe, this is why I wish you have more time to review my code! Done!